django/loader/service.py

```diff
@@ -41,20 +43,28 @@ def load(
     )
     resolved_fhir_instance = binder.resolve_references(fhir_object, analysis.reference_paths)
 
+    logger.debug({"message": "Writing document to mongo", "resource_id": resource_id})
+    load_response = loader.load(
+        resolved_fhir_instance,
+        resource_type=resolved_fhir_instance["resourceType"],
+    )
+
+    # A bit ugly, DuplicateError should have had code as a class attribute
+    if isinstance(load_response, OperationOutcome) and load_response.issue[0].code == DuplicateError("").code:
+        # We can have duplicates error if some kafka events are processed
+        # several times (this can happen if a consumer takes too much time to
+        # process the event for instance). We want to avoid deleting all the
+        # kafka topics too soon if this happens.
+        return
```

Comment:
after discussing this IRL (also with @tevariou), we agree on the fact that this is "hacky" and that we could find an actual solution for this.
- A way to mitigate (but not solve) this would be to increase `max.poll.interval.ms` consumer setting.
- We should investigate if the consumer is notified when the consumer takes longer than `max.poll.interval.ms` to process the event (between two calls to `.poll()`) which would make kafka re-send the event to a consumer later, assuming that the event was not successfully processed.

Reply:
Even if we consider that there is a way to know if a previously processed message was sent again, what do we do with this information?